### PR TITLE
Refactoring and improving the engine side bar

### PR DESCRIPTION
**Pull request by madoar**
This PR moves the sidebar code from the `EnginesView` to its own class `EngineSideBar`.
In addition it introduces the `LeftToggleGroup` class in the engines sidebar to show the different engine categories.

### CODE REVIEW

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/engines/EngineSideBar.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    private CheckBox notInstalledCheck;
+
+    private CombinedEnginesFilter currentFilter;
+
+    private Consumer<CombinedEnginesFilter> onApplyFilter = (filter) -> {};
+
+    private Consumer<EngineCategoryDTO> onCategorySelection;
+    private Runnable onAllCategorySelection;
+
+    public EngineSideBar() {
```
Comment (plata) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/engines/EngineSideBar.java]: some Javadoc please ;)

**Comment by plata** [comment]
Does the `LeftToggleGroup` change the look?

**Comment by madoar** [comment]
It changes it in some small aspects. The old (current) implementation uses `LeftButton`s for the engine selection in the left sidebar. The new implementation, which uses the `LeftToggleGroup`, uses `LeftToggleButton`s. These `LeftToggleButton`s change their background color after they are selected, see the "Apps" tab to see what I mean.
grafik

**Comment by plata** [comment]
Ok. Yes makes sense to use toggle buttons.

**Pull request merged**